BDArmory/FX/ExplosionFX.cs

```diff
@@ -788,6 +793,10 @@ private void ExecutePartBlastEvent(PartBlastHitEvent eventToExecute)
                                 else
                                 {
                                     damage = part.AddExplosiveDamage(blastInfo.Damage, Caliber, ExplosionSource, dmgMult);
+                                    if (part == hitpart && part.name.ToLower().Contains("armor")) //deal armor damage to armor panel, since we didn't do that earlier
```

Comment:
It might be worth benchmarking `part.name.ToLower().Contains("armor")` vs a hash lookup of the part name and making a `IsArmorPart` function similar to the `ProjectileUtils.IsIgnoredPart` function if there's a significant difference.
I'll try to test this later tonight to see if it's significant or not.

Reply:
Yeah, the HashSet lookup is about 4.5 times faster:
```csharp
var armourParts = PartLoader.LoadedPartsList.Select(p => p.partPrefab.partInfo.name).Where(name => name.ToLower().Contains("armor")).ToHashSet();
Debug.Log($"DEBUG Armour parts in game: " + string.Join(", ", armourParts));
int N = 1 << 24;
int count = 0;
var tic = Time.realtimeSinceStartup;
for (int i = 0; i < N; ++i)
    count += armourParts.Contains("BD.PanelArmor") ? 1 : 0;
var dt = Time.realtimeSinceStartup - tic;
Debug.Log($"DEBUG HashSet lookup took {dt/N:G3}s");
count = 0;
var armourPart = "BD.PanelArmor";
tic = Time.realtimeSinceStartup;
for (int i = 0; i < N; ++i)
    count += armourPart.ToLower().Contains("armor") ? 1 : 0;
dt = Time.realtimeSinceStartup - tic;
Debug.Log($"DEBUG SubStr lookup took {dt/N:G3}s");
```
gave
```
[LOG 23:51:01.902] DEBUG Armour parts in game: BD.PanelArmorTri, BD.PanelArmor, BD0.5x0.5panelArmor, BD0.5x0.5slopeArmor, BD1.5x0.5panelArmor, BD12x4panelArmor, BD16x4panelArmor, BD1x0.5panelArmor, BD1x0.5slopeArmor, BD1x1panelArmor, BD1x1slopeArmor, BD2x0.5panelArmor, BD2x1panelArmor, BD2x1slopeArmor, BD2x2panelArmor, BD2x2slopeArmor, BD3x1panelArmor, BD4x1panelArmor, BD4x2panelArmor, BD4x2slopeArmor, BD4x4panelArmor, BD4x4slopeArmor, BD6x2panelArmor, BD8x2panelArmor, BD8x4panelArmor, BD8x4slopeArmor, BD1x0.5ReactiveArmor
[LOG 23:51:50.917] DEBUG HashSet lookup took 4.56E-08s
[LOG 23:55:31.930] DEBUG SubStr lookup took 2.06E-07s
```

Reply:
I've added a `ProjectileUtils.IsArmorPart(Part part)` function for this and added the `"flagPart"` part names to the `IgnoredPartNames` HashSet in commit 2445af11 in dev.

Reply:
Pulled your commit and changed over the instances of part.Contains("armor") with isArmorPart(part) in the relevant secions of code; also added "conformaldecals" along with flags to the ignored parts hash, since it seemed apropos.